### PR TITLE
rgw/multisite: set versioning status in archive zone upon bucket creation

### PR DESCRIPTION
**Pull request by smanjara**
https://tracker.ceph.com/issues/63296

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_bucket.cc [hunk 1]
```diff
@@ -2574,8 +2574,14 @@ int RGWMetadataHandlerPut_BucketInstance::put_check(const DoutPrefixProvider *dp
       }
     }
     bci.info.layout.current_index.layout.type = rule_info.index_type;
+
+    // set versioning on bucket during creation on archive zone
+    if (bihandler->driver->get_zone()->get_tier_type() == "archive") {
+      bci.info.flags = BUCKET_VERSIONED;
```
[the remark this answers is not in the document]
Comment (smanjara) [reply src/rgw/driver/rados/rgw_bucket.cc:2580]: I think you're right. I just assumed that info.flags is always reserved for versioning and that we want to set it to nothing else but versioned on archive site.

File: src/rgw/driver/rados/rgw_bucket.cc [hunk 2]
```diff
@@ -2574,8 +2574,14 @@ int RGWMetadataHandlerPut_BucketInstance::put_check(const DoutPrefixProvider *dp
       }
     }
     bci.info.layout.current_index.layout.type = rule_info.index_type;
+
+    // set versioning on bucket during creation on archive zone
+    if (bihandler->driver->get_zone()->get_tier_type() == "archive") {
+      bci.info.flags = (bci.info.flags & ~BUCKET_VERSIONS_SUSPENDED) | BUCKET_VERSIONED;
+    }
```
Comment (cbodley) [inline src/rgw/driver/rados/rgw_bucket.cc:2581]: since this is repeated in both if/else blocks now, can we move it out so it's not repeated?

Comment (smanjara) [reply src/rgw/driver/rados/rgw_bucket.cc]: oops

**Review by cbodley — APPROVED** [review]
👍

**Comment by cbodley** [comment]
any idea when this stopped working? does https://tracker.ceph.com/issues/63296 need to track backports?

**Comment by smanjara** [comment]
> any idea when this stopped working? does https://tracker.ceph.com/issues/63296 need to track backports?

looks like it's always been this way. I never noticed but QE reported it recently.

**Comment by cbodley** [comment]
passed qa after rerun in https://pulpito.ceph.com/cbodley-2023-11-03_18:34:54-rgw-wip-cbodley-testing-distro-default-smithi/

**Comment by cbodley** [comment]
jenkins test api

**Pull request merged**